### DISCUSSION TITLE
Adding new categories: problem with deploy

### DISCUSSION THREAD

**Discussion by kenara** [microfeed/microfeed#127] [Q&A] [answered]
I'm adding items through the API and I need to add a few more categories so that the prefix of the returned `media_url` can be identified for my database. 

In `Constants.js`, I added the new categories to `ENCLOSURE_CATEGORIES`, `ENCLOSURE_CATEGORIES_DICT`, and `SUPPORTED_ENCLOSURE_CATEGORIES` and I added the new categories to the array of included categories on line 29 of `index.jsx`.

Saving the changes, my repo was redeployed automatically. But I must have missed something because when I tried to add an item from one of the new categories, I got an error:

 `{'error': 'Invalid category: card. Category must be one of: image, audio, video, document'}`

That error comes from index.jsx that I changed, and the workflow runs are marked as successful. Where did I go wrong?

**Comment by wenbinf** [comment]
> I'm adding items through the API and I need to add a few more categories so that the prefix of the returned `media_url` can be identified for my database.

For this use case, I'd first explore no-code solution -
For example, you can use [Cloudflare Page Rules](https://developers.cloudflare.com/rules/page-rules/) to rewrite urls to fit your needs. 

> {'error': 'Invalid category: card. Category must be one of: image, audio, video, document'}

with error message in a open source project, I'd first search in the code base and see where the error comes from - 
https://github.com/search?q=repo%3Amicrofeed%2Fmicrofeed%20Category%20must%20be%20one%20of&type=code

You'll have to make changes here: https://github.com/microfeed/microfeed/blob/31023e08361741fe843f8642f9f7ef9bf42cb9fe/functions/api/media_files/presigned_urls/index.jsx#L29

**Reply by kenara** [reply]
I made the change in index.jsx but as it wasn't reflected in the error message, I figured something was up with the deploy. Thanks for your response and sorry for the noob issue..

**Reply by kenara** [reply]
Also, for now my new categories are only useful to know the content type behind the media_url, I will need to differentiate the types when displaying them on a web page. In the end, it didn't take much to add those new categories.

**Comment by kenara** [comment]
All right, got it... although Github automatically 'deploys' the changed code, it's not sent to Cloudflare except by visiting ´Deploy to Cloudflare Pages ´ under Actions and running the workflow. Not sure what Github is doing when it says 'deploy'...

**Comment by wenbinf** [comment, marked as answer]
> All right, got it... although Github automatically 'deploys' the changed code, it's not sent to Cloudflare except by visiting ´Deploy to Cloudflare Pages ´ under Actions and running the workflow. Not sure what Github is doing when it says 'deploy'...

Deployment must be manually triggered here https://github.com/kenara/microfeed/actions/workflows/deploy.yml
